tests/zfs-tests/tests/functional/delegate/delegate_common.kshlib

```diff
@@ -425,6 +428,49 @@ function verify_fs_receive
 	return 0
 }
 
+function verify_fs_receive_append
+{
+	typeset user=$1
+	typeset perm=$2
+	typeset fs=$3
+
+	typeset dtst
+	typeset stamp=${perm}.${user}.$RANDOM
+	typeset newfs=$fs/newfs.$stamp
+	typeset bak_user=$TEST_BASE_DIR/bak.$user.$stamp
+
+	log_must zfs create $newfs
+	typeset datasets="$newfs"
+
+	for dtst in $datasets ; do
+
+		typeset dtstsnap=$dtst@snap.$stamp
+		log_must zfs snapshot $dtstsnap
+
+		log_must eval "zfs send $dtstsnap > $bak_user"
+		log_must_busy zfs destroy -rf $dtst
+
+		log_must zfs allow $user create,mount,canmount $fs
+		user_run $user eval "zfs receive -o canmount=off -F $dtst < $bak_user"
```

Comment:
Could/chould we check for error here and below?

Reply:
I am unsure, as the receive test (which I copied) does not check for such errors.

Reply:
I haven't looked how to do it, but I find weird that we verify command error not by its explicit exit status, but by expected result of its work.  Would be good to check both.